### PR TITLE
Update Rust toolchain to `1.34.0`.

### PR DESCRIPTION
**Pull request by kaikai1024**
### Description of the Change

Update Rust toolchain to `1.34.0`.

* [ ] TryInto & TryFrom *Maybe nex pr*

### Applicable Issues

#458

### CODE REVIEW

File: cita-executor/core/src/state/account.rs [hunk 1]
```diff
@@ -271,7 +272,7 @@ impl Account {
     /// Verify value proof of the trie's storage at `key`.
     pub fn verify_value_proof(&self, key: &H256, proof: &[Bytes]) -> Option<H256> {
         trie::triedb::verify_value_proof(key, self.storage_root, proof, ::rlp::decode)
-            .map(|v: U256| v.into())
+            .map(&Into::into as &Fn(U256) -> H256)
```
Comment (kaikai1024) [inline cita-executor/core/src/state/account.rs]: 这里 clippy 提示直接修改： `.map(Into::into)`
编译会报错，需要制定类型

File: env.sh [hunk 2]
```diff
@@ -80,10 +80,10 @@ test -t 1 && USE_TTY="-t"
 # Condition `daemon` to run daemon.
 if [ "$3" == "--daemon"  ]; then
     set "${@:1:2}" "${@:4}"
-    docker exec -d ${CONTAINER_NAME} /bin/bash -c "/usr/bin/gosu ${USER_NAME} $* >/dev/null 2>&1"
+    docker exec -d ${CONTAINER_NAME} /bin/bash -c "/usr/sbin/gosu ${USER_NAME} $* >/dev/null 2>&1"
```
Comment (kaikai1024) [inline env.sh]: the new path when use apt to install it.

File: cita-chain/core/src/libchain/chain.rs [hunk 3]
```diff
@@ -1124,7 +1124,7 @@ impl Chain {
                         receipts.len(),
                         hashes.len()
                     );
-                    assert!(false);
+                    panic!();
```
Comment (yangby-cryptape) [inline cita-chain/core/src/libchain/chain.rs]: Add a message in `panic!(...)` or just use `unreadable!()`.

Comment (kaikai1024) [reply cita-chain/core/src/libchain/chain.rs]: ok

File: cita-auth/src/dispatcher.rs [hunk 4]
```diff
@@ -155,8 +155,8 @@ impl Dispatcher {
         let pool = self.txs_pool.borrow();
         ids.iter()
             .map(|id| pool.get(id).cloned())
-            .filter(|tx| tx.is_some())
-            .map(|tx| tx.unwrap())
+            .filter(Option::is_some)
```
Comment (yangby-cryptape) [inline cita-auth/src/dispatcher.rs]: Ref: [`fn filter_map<B, F>(self, f: F) -> FilterMap<Self, F> where F: FnMut(Self::Item) -> Option<B>`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map)

Comment (kaikai1024) [reply cita-auth/src/dispatcher.rs]: Nice

**Pull request closed without merging**